### PR TITLE
Fix Testing GitHub Actions

### LINKED ISSUE #284: Add/Separate GitHub PR Check into distinct Unit and Functional checks
Add a PR check in GitHub for DSPO to run the new DSPO Unit Test suite.  This likely already is happening, but is running *both* the unit and functional tests at the same time, so seperating them into two distinct checks will allow better granularity when debugging

### PR DESCRIPTION
**Pull request by gmfrasca**
## The issue resolved by this Pull Request:
Resolves #284 

## Description of your changes:
- Remove old pre-commit go-unit-test check (doesn't work with new tags used in tests)
- Add Unit Test GitHub Action
- Add Functional Test GitHub Action

## Testing instructions
Verify the Checks on this PR pass :) 

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by DharmitD — COMMENTED** [review]
/lgtm

File: .github/workflows/functests.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Run Functional Tests
```
Comment (HumairAK) [inline .github/workflows/functests.yml:1]: can we omit `Run` from the title in both, and just have it show up as "Functional Tests" and "Unit Tests"

**Pull request merged**